interview-questions.md

```diff
@@ -8,62 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: The parameter can happen when a Function is created. The parameter goes inside of the parenthesis after the name of the function. The parameter is a placeholder that is declared in the parenthesis. The argument is that the data is passed onto the function upon it's invocation. Unlike the parameter, the arguments do not have names. Each argument is an expression, which can contain zero or more variables, constants, and literals. The argument can be different every time that it is called. 
 
-  Researched answer:
+  Researched answer: While doing my reseach, I found that when a function is created it can have parameters. The parameters goes inside of the parenthesis and only need to be defined in a function once. On the other hand, the argument can be different each time a function is invoked. The argument will get matched up with the parameter. So parameters are the names listed in the function and arguments are the actual values passed onto the function.  
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The javaScript built in method .map() takes the value, index, and array as parameters. Value is always required while using map, while index and array are optional and only used if needed.
+
+  Researched answer: Map is a function that can iterate through an array and return a new array with updated values of the same length. It takes predefined parameters such as value, index, and array. The index and array are optional and only used when the programmer needs it.
 
-  Researched answer:
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: The difference between map and filter are that map functions do not change the original array, but instead iterate through the array and returns a new array of the same lengtht. The filter function also will not change the original array, as it will iterate through the array, but it will return ONLY the values that are true for the new array and the value that satisfy the past conditions.  
 
-  Researched answer:
+  Researched answer: Looking at the syllabus, I found that even though map and filter contain the same parameters (value, index, array), the outcomes are different. Map is a function that is going to be useful when the programmer wants to iterate through each item in an array and return a new array with updated values with the same length. Filter function, on the other hand, will lopp through an array and will make a decision to bring back only the group that satisfy the stated condition. 
```

Comment:
Certainly!  Filter will have a built-in if-else statement, whereas map can perform logic on all values.